### PR TITLE
(DOCSP-27664) Update top Data API landing to reflect new positioning / frontend access

### PR DESCRIPTION
**Pull request by nlarew**
## Pull Request Info

- Updates Data API landing page to focus more on browser use case + expand conceptual/technical explanations.
- Moves data formats from the landing page to a separate sub-page
- Note: this is rebased on top of https://github.com/mongodb/docs-app-services/pull/336

### Jira

- (DOCSP-27664) Update top Data API landing to reflect new positioning / frontend access

### Staged Changes

- [Data API](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/api-auth-2/data-api/)
- (Moved) [Data API > Data Formats](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/api-auth-2/data-api/data-formats/)

### CODE REVIEW

File: source/data-api.txt
Comment (mongodben) [inline source/data-api.txt:37]:
```suggestion
to work with the Data API. Instead, you send standard HTTPS requests like the
```

File: source/data-api.txt
Comment (mongodben) [inline source/data-api.txt:64]: i don't think it's smthn to address in this PR, but i find it kinda funny that we're calling both generated endpoints and custom endpoints 'the data api', but then also the generated endpoints are called 'the data api'.

probably this is smthn that extends to product/product marketing. but worth bringing up imo.

curious your thoughts.

File: source/data-api.txt
Comment (mongodben) [inline source/data-api.txt:76]:
```suggestion
- **Custom Endpoints** are app-specific API routes handled by Atlas Functions
```

File: source/data-api.txt
Comment (mongodben) [inline source/data-api.txt:93]: to call web apps 'unsafe' rings a little funky to me.

i wonder if there's a better word that we could use here. perhaps smthn like: 

```suggestion
access data in Atlas from potentially vulnerable clients like web apps.
```

(got that suggestion w help of chat gpt 😉)

File: source/data-api.txt
Comment (mongodben) [inline source/data-api.txt:112]:
```suggestion
   your App. This means that users can only access data they're allowed
   to read and write. Requests fail if they include an invalid write
```

File: source/data-api.txt
Comment (nlarew) [reply source/data-api.txt:64]: Yeah it's silly IMO but that's what we're calling them for now. I have some ideas on how we could be clearer that I've brought up with Sumedha but I'll reintroduce them with PM again.

File: source/data-api.txt
Comment (mongodben) [inline source/data-api.txt:119]: the last sentence reads a little funny to me b/c it seems to assign agency to the request. consider rephrase to assign agency to developer: 

```suggestion
   return from a custom endpoint. In the request, you can choose to receive the
   response in either :ref:`JSON or EJSON format
   <data-api-data-formats>`.
```

File: source/data-api.txt
Comment (mongodben) [inline source/data-api.txt:133]:
```suggestion
- You want to integrate Atlas data access into a federated API gateway.
```

File: source/data-api.txt
Comment (mongodben) [inline source/data-api.txt:136]: ref to realm sdks landing page here? https://www.mongodb.com/docs/realm/

tho on reading the landing page, it really positions realm as a mobile DB, rather than a mobile DB + client to access App Services, so reader might be confused if they're not familiar w realm and wind up there, so 🤷 

curious your thoughts here. 

obvi out of scope for this PR, but maybe we should have a page about realm on the App Services docs explaining what Realm SDKs are, and how they related to App Services.

File: source/data-api.txt
Comment (mongodben) [inline source/data-api.txt:12]: this page is an excellent product overview!

File: source/data-api/data-formats.txt
Comment (mongodben) [inline source/data-api/data-formats.txt:16]:
```suggestion
encoding. BSON is efficient for computers but is not human readable,
so you can't work with it directly.
```

File: source/data-api/data-formats.txt
Comment (mongodben) [inline source/data-api/data-formats.txt:13]: maybe link word BSON to docs? 

```suggestion
MongoDB stores data in a format called `BSON <https://bsonspec.org/>`__,
which is similar to a JSON
```

File: source/data-api/data-formats.txt
Comment (mongodben) [inline source/data-api/data-formats.txt:31]: doesn't need to be in this PR, but would be helpful to point users to docs on how to work w EJSON from client code.

**Review by mongodben — APPROVED** [review]
nice work here! like the new landing page a lot. clearly explains the value prop. 

couple small things and few questions throughout but nothing blocking.

File: source/data-api.txt
Comment (mongodben) [inline source/data-api.txt:144]: just noticed this link is broken. fix before merge.

File: source/data-api.txt
Comment (drewdipalma) [inline source/data-api.txt:32]: Nit: I think this might help set the stage better either combined with the first sentence or as a second sentence.  Feels like we're splitting the definition in two here.

File: source/data-api.txt
Comment (drewdipalma) [inline source/data-api.txt:139]: I would like us to be more specific here and add above the bulleted list.  Something along the lines of "Using the MongoDB Drivers to build on top of Atlas is still the general recommended approach (especially when connecting from a Server) as it will be more performant and flexible."

File: source/data-api.txt
Comment (drewdipalma) [inline source/data-api.txt:144]: Another bullet that we might consider adding to this list – 

- "You are looking to create a new feature and would prefer starting your development on the client side and then moving to create / refine the API layer when your requirements are refined."

**Review by drewdipalma — APPROVED** [review]
Approving with some minor comments

File: source/data-api/data-formats.txt
Comment (nlarew) [reply source/data-api/data-formats.txt:32]: Don't think we actually have a great article on this. Not sure if we want it in this section or elsewhere but would be great to have.

File: source/data-api.txt
Comment (nlarew) [reply source/data-api.txt:153]: > on reading the landing page, it really positions realm as a mobile DB, rather than a mobile DB + client to access App Services, so reader might be confused if they're not familiar w realm and wind up there

Link could be helpful. I think the paragraph that linked them there has enough context that some/most SDKs have an App Services integration.

> obvi out of scope for this PR, but maybe we should have a page about realm on the App Services docs explaining what Realm SDKs are, and how they related to App Services

Agreed - I thought we at least had this on the introduction page but we don't even really have it there, just a couple mentions of the SDKs in broader contexts.

**Pull request merged**
